server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -1174,6 +1174,35 @@ public TSExecuteStatementResp insert(TSInsertionReq req) {
     }
   }
 
+  @Override
+  public TSStatus insertRowInBatch(TSInsertInBatchReq req) {
+    if (!checkLogin()) {
+      logger.info(INFO_NOT_LOGIN, IoTDBConstant.GLOBAL_DB_NAME);
+      return new TSStatus(getStatus(TSStatusCode.NOT_LOGIN_ERROR));
+    }
+
+    TSStatus status = null;
+    for (int i = 0; i < req.deviceIds.size(); i++) {
+      InsertPlan plan = new InsertPlan();
+      plan.setDeviceId(req.getDeviceIds().get(i));
+      plan.setTime(req.getTimestamps().get(i));
+      plan.setMeasurements(req.getMeasurementsList().get(i).toArray(new String[0]));
+      plan.setValues(req.getValuesList().get(i).toArray(new String[0]));
+      status = checkAuthority(plan);
```

Comment:
this check is not needed each time

Reply:
Maybe the second insertion's path is different from the first insertion. So we have to check it every time